### PR TITLE
Bindings

### PR DESCRIPTION
**Pull request by frankschlegel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -683,6 +684,52 @@ subject.send(completion: .failure(.someError))
 .finished
 ```
 
+### bind
+
+A binding is a subscription between a publisher and an object that implements the `BindingTarget` protocol. The lifetime of the subscription is in this case bound to the lifetime of the target—the subscription will be cancelled when the binding target gets deallocated.
+
+An object can conform to `BindingTarget` by providing a providing a "bag" to store the cancellables of its bindings:
```
Comment (maikz) [inline README.md]: One providing to many

File: Sources/Bindings/BindingTarget.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+///     aPublisher.bind(to: self, keyPath: \.property)
+///
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public protocol BindingTarget: AnyObject {
+    /// A "bag" to store subscriptions in that should be cancelled
+    /// when `self` is deallocated.
+    var cancellables: Set<AnyCancellable> { get set }
+}
+
+#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
```
Comment (maikz) [inline Sources/Bindings/BindingTarget.swift]: Aren't those all the possible OS?

Comment (frankschlegel) [reply Sources/Bindings/BindingTarget.swift]: There's also Linux 😉 (and potentially more platforms coming). But yes, I could probably use `#canImport` here.

File: Sources/Operators/Bind.swift [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            if let target = target { receiveCompletion(target, completion) }
+        }, receiveValue: { [weak target] element in
+            if let target = target { receiveValue(target, element) }
+        }).store(in: &target.cancellables)
+    }
+}
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publisher where Self.Failure == Never {
+    /// "Binds" the publisher to the target, i.e. the target receives values events
```
Comment (maikz) [inline Sources/Operators/Bind.swift]: ~~events~~

File: Sources/Operators/Bind.swift [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    /// and ensures that the subscription is cancelled with the target is deallocated.
```
Comment (maikz) [inline Sources/Operators/Bind.swift]:
```suggestion
    /// and ensures that the subscription is cancelled when the target is deallocated.
```

File: Sources/Operators/Bind.swift [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /// "Binds" the publisher to the target and assigns new values to a property.
+    /// The subscription created this way is cancelled when the target is deallocated.
+    ///
+    /// Using this method will _not_ accidentally create any reference cycle between
+    /// the subscription and the target, in contrast to Combine's `assign(to:on:)`.
+    /// See [this forum discussion](https://forums.swift.org/t/does-assign-to-produce-memory-leaks/29546)
+    /// - Parameters:
+    ///   - keyPath: A key path that indicates the property to assign.
```
Comment (maikz) [inline Sources/Operators/Bind.swift:54]: maybe switch the ordering as it is in the function

File: Sources/Operators/Bind.swift [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    ///   - target: The object that contains the property. The subscriber assigns the object’s property every time it receives a new value.
```
Comment (maikz) [inline Sources/Operators/Bind.swift]:
```suggestion
    ///   - target: The object that contains the property. The subscriber assigns the object's property every time it receives a new value.
```

File: Sources/Operators/Bind.swift [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        sink { [weak target] element in
+            target?[keyPath: keyPath] = element
+        }.store(in: &target.cancellables)
+    }
+}
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publisher where Self.Output == Void {
+    /// "Binds" the publisher to the target, i.e. the target receives values and completion events
+    /// and ensures that the subscription is cancelled with the target is deallocated.
```
Comment (maikz) [inline Sources/Operators/Bind.swift]:
```suggestion
    /// and ensures that the subscription is cancelled when the target is deallocated.
```

File: Sources/Operators/Bind.swift [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        }, receiveValue: { [weak target] in
+            if let target = target { receiveValue(target) }
+        }).store(in: &target.cancellables)
+    }
+}
+
+@available(OSX 10.15, iOS 13.0, tvOS 13.0, watchOS 6.0, *)
+public extension Publisher where Self.Output == Void, Self.Failure == Never {
+    /// "Binds" the publisher to the target, i.e. the target receives values events
+    /// and ensures that the subscription is cancelled with the target is deallocated.
```
Comment (maikz) [inline Sources/Operators/Bind.swift]:
```suggestion
    /// and ensures that the subscription is cancelled when the target is deallocated.
```

**Review by maikz — CHANGES_REQUESTED** [review]
See commented lines

**Comment by frankschlegel** [comment]
Thanks, @maikz! I incorporated your remarks.

**Review by maikz — APPROVED** [review]

**Pull request merged**